src/core/load-fragment.c

```diff
@@ -4727,20 +4727,20 @@ int config_parse_exec_directories(
                         continue;
                 }
 
+                if (!isempty(dest) && streq(lvalue, "ConfigurationDirectory")) {
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "Additional parameter is not supported for ConfigurationDirectory, ignoring: %s", tuple);
+                        continue;
+                }
```

Comment:
i'd drop this, as mentioned.

Reply:
Expanding the symlinks usage is not really related to adding read-only support, so I'd much prefer to leave that as-is for now